### PR TITLE
[kotlin-client][jvm-spring-*] Fix runtime error in endpoints of type Unit

### LINKED ISSUE #17576: [BUG][kotlin-client][jvm-spring-*] Endpoints with result type Unit will crash
##### Description

Calling endpoints with a result type Unit will crash as the field used is undefined.

##### Workaround

The current workaround is to use the WithHttpInfo variants of the endpoints.

##### openapi-generator version

7.2.0, since inception of the libraries.

##### Related issues/PRs

To find such bugs during development stage I wil add an Echo API test. E.g. https://github.com/OpenAPITools/openapi-generator/pull/16729

##### Suggest a fix

I will create a PR later this week.

### PR DESCRIPTION
**Pull request by stefankoppier**
Fixes #17576. 

I also added an echo-api test which showed two problems:
1. The known problems of anyOf/allOf support.
2. Generation of default values for enum items are incorrect.

These two problems were removed from the test by commenting out the corresponding parts in the api.

@jimschubert (2017/09) [❤️](https://www.patreon.com/jimschubert), @dr4ke616 (2018/08) @karismann (2019/03) @Zomzog (2019/04) @andrewemery (2019/10) @4brunu (2019/11) @yutaka0m (2020/03) @stefankoppier (2022/06)

### CODE REVIEW

File: bin/configs/kotlin-jvm-spring-3-restclient-echo-api.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+generatorName: kotlin
+outputDir: samples/client/echo_api/kotlin-jvm-spring-3-restclient
```
Comment (wing328) [inline bin/configs/kotlin-jvm-spring-3-restclient-echo-api.yaml:2]: can you please also add the new folder to `.github/workflows/samples-kotlin-*` files?

Comment (stefankoppier) [reply bin/configs/kotlin-jvm-spring-3-restclient-echo-api.yaml:2]: Added

**Comment by wing328** [comment]
> I also added an echo-api test which showed two problems:

I've added the echo server response parser via https://github.com/OpenAPITools/openapi-generator/pull/17887

but I need to comment out the JSON response test (e.g. Pet) as the payload contains an additional line containing just `cc` right before the json payload: https://github.com/OpenAPITools/openapi-generator/pull/17887/files#diff-2f798b020be93c6a273de6c5e8c6656044e182759b5d715bd58f8d63801bef27R27

@stefankoppier may I know if you've any clue about this issue? did you get issues when sending JSON payload to server using the kotlin client?

**Comment by stefankoppier** [comment]
@wing328 messaged you on slack.

**Pull request merged**